apps/api/src/domain/models/media-item/commands/media-item-created.event.ts

```diff
@@ -0,0 +1,22 @@
+import { AggregateId } from '../../../types/AggregateId';
+import BaseDomainModel from '../../BaseDomainModel';
+import { IEvent } from '../../shared/events/interfaces/event.interface';
+import { EventRecordMetadata } from '../../song/commands/song-created.event';
+import { CreateMediaItem } from './create-media-item.command';
+
+export class MediaItemCreated extends BaseDomainModel implements IEvent {
+    type = 'MEDIA_ITEM_CREATED';
+
+    meta: EventRecordMetadata;
+
+    constructor(command: CreateMediaItem, eventId: AggregateId) {
```

Comment:
I wonder what it would look like to inject the ID generator into the event CTOR. Does it make any difference? If we inject it into a `BaseEvent` class for once and all, we can call this in a `setMeta` method and make sure that `generate` and `reserve` are called atomically without the need to do this elsewhere. 

Should this be the event's responsibility?